integration/experiment/machine.py

```diff
@@ -46,10 +46,15 @@ def load(self):
         try:
             with open(self.path) as fid:
                 data = json.load(fid)
-                if 'signals' in data and 'topo' in data:
+                try:
                     self.signals = data['signals']
                     self.topo = data['topo']
-                else:
+                    try:
+                        self.meminfo = data['meminfo']
+                    except:
+                        #  :( :(
```

Comment:
I think it isn't so sad to support backward compatibility.  Might want a different comment.